client/my-sites/stats/features/modules/stats-locations/stats-locations.tsx

```diff
@@ -303,7 +303,7 @@ const StatsLocations: React.FC< StatsModuleLocationsProps > = ( {
 
 	return (
 		<>
-			{ ! shouldGateStatsModule && siteId && statType && (
+			{ ! shouldGate && ! supportsLocationsStatsFeature && geoMode === 'country' && (
 				<QuerySiteStats statType={ statType } siteId={ siteId } query={ query } />
```

Comment:
QuerySiteStats is the query component that fetches the legacy country-views API.
We should still use it when a site doesn't support the Locations stats feature, to render the countries list. PR related: https://github.com/Automattic/wp-calypso/pull/99278